### PR TITLE
[clang] Fix static analyzer concerns

### PR DESCRIPTION
**Pull request by Fznamznon**
It seems in checkOpenMPIterationSpace `OrderedLoopCountExpr` can also be null, so check before dereferencing.

### CODE REVIEW

File: clang/lib/Sema/SemaOpenMP.cpp [hunk 1]
```diff
@@ -9195,7 +9195,7 @@ static bool checkOpenMPIterationSpace(
         SemaRef.Diag(CollapseLoopCountExpr->getExprLoc(),
                      diag::note_omp_collapse_ordered_expr)
             << 0 << CollapseLoopCountExpr->getSourceRange();
-      else
+      else if (OrderedLoopCountExpr)
```
Comment (alexey-bataev) [inline clang/lib/Sema/SemaOpenMP.cpp:9198]: I think better to fix the logic in line 9189:
```
if ((CollapseLoopCountExpr && OrderedLoopCountExpr) || (!CollapseLoopCountExpr && !OrderedLoopCountExpr))
```

Comment (Fznamznon) [reply clang/lib/Sema/SemaOpenMP.cpp:9198]: The problem with this approach is that both `OrderedLoopCountExpr` and `CollapseLoopCountExpr` will be dereferenced on lines 9192 and 9193 if they're both are `nullptr`s.

Comment (alexey-bataev) [reply clang/lib/Sema/SemaOpenMP.cpp:9198]: Yep, right

**Review by alexey-bataev — APPROVED** [review]

**Comment by Fznamznon** [comment]
> LLVM Buildbot has detected a new failure on builder openmp-offload-amdgpu-runtime running on omp-vega20-0 while building clang at step 7 "Add check check-offload".

That seems to be failing the same way for some other PRs.

**Pull request merged**